src/main/java/seedu/bookmark/model/ModelManager.java

```diff
@@ -180,6 +202,7 @@ public void updateFilteredBookList(Predicate<Book> predicate) {
 
     @Override
     public void sortFilteredBookList(Comparator<Book> comparator) {
+        this.comparator = comparator;
         this.library.sortBooks(comparator);
     }
```

Comment:
Could you add this code to the end of the method 
`historyManager = historyManager.addNewState(State.createState(library, userPrefs, filteredBooks.getPredicate()));`
to make the sort command undoable?